Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/volunteer/VolunteerFragment.kt

```diff
@@ -0,0 +1,100 @@
+package com.mentalmachines.droidcon_boston.views.speaker
+
+
+import android.content.Context
+import android.content.DialogInterface
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.support.v7.app.AlertDialog
+import android.support.v7.widget.LinearLayoutManager
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.ValueEventListener
+import com.mentalmachines.droidcon_boston.R
+import com.mentalmachines.droidcon_boston.R.string
+import com.mentalmachines.droidcon_boston.data.FirebaseDatabase.VolunteerEvent
+import com.mentalmachines.droidcon_boston.firebase.FirebaseHelper
+import eu.davidea.flexibleadapter.FlexibleAdapter
+import eu.davidea.flexibleadapter.common.FlexibleItemDecoration
+import kotlinx.android.synthetic.main.volunteer_fragment.volunteer_recycler
+
+
+class VolunteerFragment : Fragment(), FlexibleAdapter.OnItemClickListener {
+
+    private val firebaseHelper = FirebaseHelper.instance
+    private lateinit var volunteerAdapter: FlexibleAdapter<VolunteerAdapterItem>
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        super.onCreateView(inflater, container, savedInstanceState)
+        return inflater.inflate(R.layout.volunteer_fragment, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        fetchDataFromFirebase()
+    }
+
+    private fun fetchDataFromFirebase() {
+        firebaseHelper.volunteerDatabase.orderByChild("name").addValueEventListener(object : ValueEventListener {
```

Comment:
I don't think there is a "name" field in the volunteers data...I could add the concat'd first/last name if you want

Reply:
No. Don't I think it should be firstName. Let me check that.